src/main/webapp/components/utils/getField.js

```diff
@@ -0,0 +1,12 @@
+/**
+  * Useful for extracting a field from an object, with a fallback (defaultValue).
+  * Useful where "undefined", is not a desired result for a missing field.
+  *
+  * @param obj the object to query the fields for.
+  * @param field the target field, in string form. 
+  * @param defaultValue the value to be used if obj[field] === undefined.
+ */
+export const getField = (obj, field, defaultValue) => {
+  return obj[field] !== undefined ?
+  obj[field] : (defaultValue !== undefined ? defaultValue : null);
```

Comment:
Is this something that should be indented 2 more spaces? I'm unfortunately not familiar with JS style at all.

Reply:
[+4 spaces](https://google.github.io/styleguide/jsguide.html#formatting-indent)

But actually you should be using [clang-format](https://google.github.io/styleguide/jsguide.html#appendices-clang-format).

Reply:
@MForster Thanks for the suggestion, just ran the file through clang-format.